components/collective-navbar/NavBarCategoryDropdown.js

```diff
@@ -177,24 +171,29 @@ NavBarCategory.propTypes = {
 };
 
 const NavBarCategoryDropdown = ({ useAnchor, collective, category, isSelected, links }) => {
+  const router = useRouter();
   const displayedLinks = links.filter(link => !link.hide);
   return (
     <CategoryDropdown trigger="hover" tabIndex="-1">
-      <CategoryContainer
-        mr={[0, null, 3]}
-        isSelected={isSelected}
-        {...getLinkProps(useAnchor, collective, category)}
-        onClick={e => {
-          // Remove focus to make sure dropdown gets closed
-          if (document.activeElement?.contains(e.target)) {
-            document.activeElement.blur();
-          }
-        }}
-      >
-        <Flex pt="15px" pb="14px" px={[3, 1, 3, 1]}>
-          <NavBarCategory category={category} />{' '}
-        </Flex>
-      </CategoryContainer>
+      <Scrollchor to={useAnchor ? `#category-${category}` : null}>
+        <CategoryContainer
+          mr={[0, null, 3]}
+          isSelected={isSelected}
+          onClick={e => {
+            // Remove focus to make sure dropdown gets closed
+            if (document.activeElement?.contains(e.target)) {
+              document.activeElement.blur();
+            }
+            if (!useAnchor) {
+              router.push(`/${collective.slug}/#category-${category}`);
```

Comment:
To be valid, links must have a proper `href` set. Navigating with JS is usually discouraged and results in issues like:
- Poor indexing from search engines
- Poor accessibility (e.g. middle-mouse click / open in a new tab doesn't work)

Reply:
@Betree : Thanks much for the explanation. I've re-added the `href` tags and also removed the leftover smooth-scroll related Cypress code.

Reply:
This `router.push` should not be needed if we handle href correctly

Reply:
Yes, sorry to make a bit of a mess out of this. I've removed this one as it's not needed.